erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -225,6 +225,7 @@ def on_submit(self):
 		self.update_serial_no()
 
 		if not cint(self.is_pos) == 1 and not self.is_return:
+			self.db_update()
```

Comment:
Why a `db_update` is needed here?
I am assuming this should be taken care of by this function
https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/utils.py#L341

Even if it doesn't it should be placed somewhere in this function instead of directly in the sales invoice `on_submit` trigger

Reply:
Makes sense, will push the change